crates/testing/tests/consensus_task.rs

```diff
@@ -242,6 +242,319 @@ async fn test_consensus_vote_with_permuted_dac() {
+
+    let handle = build_system_handle(5).await.0;
+    let quorum_membership = handle.hotshot.memberships.quorum_membership.clone();
+
+    let view_sync_finalize_data: ViewSyncFinalizeData<TestTypes> = ViewSyncFinalizeData {
+        relay: 10,
+        round: ViewNumber::new(10),
+    };
+
+    let mut generator = TestViewGenerator::generate(quorum_membership.clone());
+    let mut proposals = Vec::new();
+    let mut leaders = Vec::new();
+    let mut votes = Vec::new();
+    let mut vids = Vec::new();
+    let mut dacs = Vec::new();
+    for view in (&mut generator).take(2) {
+        proposals.push(view.quorum_proposal.clone());
+        leaders.push(view.leader_public_key);
+        votes.push(view.create_vote(&handle));
+        vids.push(view.vid_proposal.clone());
+        dacs.push(view.da_certificate.clone());
+    }
+
+    // Each call to `take` moves us to the next generated view. We advance to view
+    // 3 and then add the finalize cert for checking there.
+    generator.add_view_sync_finalize(view_sync_finalize_data);
+    for view in (&mut generator).take(1) {
+        proposals.push(view.quorum_proposal.clone());
+        leaders.push(view.leader_public_key);
+        votes.push(view.create_vote(&handle));
+        vids.push(view.vid_proposal.clone());
+        dacs.push(view.da_certificate.clone());
+    }
+
+    let view_1 = TestScriptStage {
+        inputs: vec![QuorumProposalRecv(proposals[0].clone(), leaders[0])],
+        outputs: vec![
+            exact(ViewChange(ViewNumber::new(1))),
+            exact(QuorumProposalValidated(proposals[0].data.clone())),
+            exact(QuorumVoteSend(votes[0].clone())),
+        ],
+        asserts: vec![is_at_view_number(1)],
+    };
+
+    let view_2 = TestScriptStage {
+        inputs: vec![
+            VidDisperseRecv(vids[1].0.clone(), vids[1].1),
+            QuorumProposalRecv(proposals[1].clone(), leaders[1]),
+            DACRecv(dacs[1].clone()),
+        ],
+        outputs: vec![
+            exact(ViewChange(ViewNumber::new(2))),
+            exact(QuorumProposalValidated(proposals[1].data.clone())),
+            exact(QuorumVoteSend(votes[1].clone())),
+        ],
+        asserts: vec![is_at_view_number(2)],
+    };
+
+    let cert = proposals[2].data.view_sync_certificate.clone().unwrap();
+    // Obtain the ViewSyncFinalizeCertificate2Recv and make sure that the vote goes through
+    let view_3 = TestScriptStage {
+        inputs: vec![
+            ViewSyncFinalizeCertificate2Recv(cert),
+            QuorumProposalRecv(proposals[2].clone(), leaders[2]),
+            VidDisperseRecv(vids[2].0.clone(), vids[2].1),
+            DACRecv(dacs[2].clone()),
+        ],
+        outputs: vec![
+            exact(ViewChange(ViewNumber::new(3))),
+            exact(QuorumProposalValidated(proposals[2].data.clone())),
+            leaf_decided(),
+            quorum_vote_send(),
+        ],
+        asserts: vec![is_at_view_number(3)],
+    };
```

Comment:
Sorry I missed this on first pass, this doesn't really test what we want.  We're voting here because the `QuorumProposal` has a QC for the previous view.  That's enough to vote without finalized cert.  Also I think we should test this works if the node doesn't receive `ViewSyncFinalizeCertificate2Recv`

Reply:
Instead, we need the QuorumProposal to have a ViewSync cert, correct? Then we can go through the proper flow?

To make this work without receiving a view sync cert, I'd need to have another cert available to be able to have the proposal be valid so I (the node) can vote on it. So I could just drop the view sync and this would likely run just fine. I'll do that in a separate test.